common/lib/common-definitions/api-report/common-definitions.api.md

```diff
@@ -4,6 +4,16 @@
 
 ```ts
 
+// @public
+export namespace Compat {
+    export interface ITelemetryBaseLogger {
```

Comment:
I did not look at the rest of the change (and if this direction is really needed and the right one), but if we need to go that route, I'd rather go for something that much more easily supports multiple versions / can withstand time test.

I.e. 
```typescript
export interface ITelemetryBaseLogger1 {...}
export interface ITelemetryBaseLogger2 {...}
export interface ITelemetryBaseLogger3 {...}

// Latest and greatest version
type ITelemetryBaseLogger = ITelemetryBaseLogger3;
```
This pattern is used across C++ SDKs :)
And it's really powerful and required in cases like this, where latest version of SDK (headers, definitions) is mostly decoupled from the version of functionality (in our case - actual packages with code) one might be consuming.

Reply:
Cool, this is exactly the feedback I was looking for. Thanks!